File: verletplus/src/utils/Colors.cs
using System;

namespace utils
{
    class Colors
    {
        // Colors
        public static readonly (int, int, int) RED = (255, 0, 0);
        public static readonly (int, int, int) GREEN = (0, 255, 0);
        public static readonly (int, int, int) BLUE = (0, 0, 255);
        public static readonly (int, int, int) WHITE = (255, 255, 255);
        public static readonly (int, int, int) BLACK = (0, 0, 0);
        public static readonly (int, int, int) YELLOW = (255, 255, 0);
        public static readonly (int, int, int) PURPLE = (255, 0, 255);
        public static readonly (int, int, int) CYAN = (0, 255, 255);
        public static readonly (int, int, int) ORANGE = (255, 165, 0);
        public static readonly (int, int, int) PINK = (255, 192, 203);

        // Pick a random color from the list
        public static (int, int, int) RandomColor()
        {
            Random random = new Random();
            (int, int, int)[] colors = { RED, GREEN, BLUE, WHITE, YELLOW, PURPLE, CYAN, ORANGE, PINK };
            return colors[random.Next(0, colors.Length)];
        }

        // Pick a random color rgb
        public static (int, int, int) RandomRGB()
        {
            Random random = new Random();
            return (random.Next(0, 255), random.Next(0, 255), random.Next(0, 255));
        }

        // Pick a random color hex
        public static string RandomHex()
        {
            Random random = new Random();
            return String.Format("#{0:X6}", random.Next(0x1000000));
        }
    }
}

File: verletplus/src/utils/colors.cpp
#include <iostream>

class Color
{
public:
    int r;
    int g;
    int b;

    Color(int r, int g, int b)
    {
        this->r = r;
        this->g = g;
        this->b = b;
    }
};

class Colors
{
public:
    // Constant colors
    Color RED = Color(255, 0, 0);
    Color GREEN = Color(0, 255, 0);
    Color BLUE = Color(0, 0, 255);
    Color YELLOW = Color(255, 255, 0);
    Color CYAN = Color(0, 255, 255);
    Color MAGENTA = Color(255, 0, 255);
    Color WHITE = Color(255, 255, 255);
    Color BLACK = Color(0, 0, 0);
    Color ORANGE = Color(255, 165, 0);
    Color PURPLE = Color(128, 0, 128);
    Color PINK = Color(255, 192, 203);

    // Generate a random rgb color
    std::tuple<int, int, int> random_rgb()
    {
        return std::make_tuple(rand() % 255, rand() % 255, rand() % 255);
    }

    // Generate a random color from the constant colors
    Color random_color()
    {
        Color colors[11] = {RED, GREEN, BLUE, YELLOW, CYAN, MAGENTA, WHITE, BLACK, ORANGE, PURPLE, PINK};
        return colors[rand() % 11];
    }
};